src/Game/AccountManagement.java

```diff
@@ -114,6 +123,222 @@ public static void register(String user, String pass) {
       }
     }
   }
+  
+  public static void forgotPass(String user) {
+	  Client.setLoginMessage("Connecting to server", "Please wait...");
+
+		try {
+			int port = Replay.connection_port;
+	        StreamUtil.initializeStream(Client.server_address, port);
+	        StreamUtil.setStreamMaxRetries(Client.maxRetries);
+
+	        Client.session_id = 0; // TODO: should read int to get session ID here, triggered by TCP handshake
+	        
+	        StreamUtil.newPacket(4);
+	        Object buffer = StreamUtil.getStreamBuffer();
+	        StreamUtil.putLongTo(buffer, Util.username2hash(user));
+	        StreamUtil.flushPacket();
+	        
+			StreamUtil.readShort(); // Unknown what data this contained, client doesn't use it.
+			int response = StreamUtil.readStream();
+			Logger.Game("Getpq response: " + response);
+			if (response == 0) {
+				Client.setLoginMessage("", "Sorry, the recovery questions for this user have not been set");
+				return;
+			}
+
+			String question;
+			int idx;
+			for (int var12 = 0; var12 < 5; ++var12) {
+				int length = StreamUtil.readStream();
+                byte[] arr = new byte[5000];
+                StreamUtil.readBytes(arr, length);
+                question = new String(arr, 0, length);
+                
+                if (question.startsWith("~:")) {
+                	question = question.substring(2);
+					idx = 0;
+
+					try {
+						idx = Integer.parseInt(question);
+					} catch (Exception var8) { }
+
+					question = recoveryQuestions[idx];
+				}
+                
+                Panel.setControlText(Client.panelRecovery, Client.controlRecoveryQuestion[var12], question);
+			}
+
+			if (Client.failedRecovery) {
+				Client.setLoginMessage("", "Sorry, you have already attempted 1 recovery, try again later");
+				return;
+			}
+
+			Client.login_screen = Client.SCREEN_PASSWORD_RECOVERY;
+			Panel.setControlText(Client.panelRecovery, Client.controlRecovery1,
+					"@yel@To prove this is your account please provide the answers to");
+			Panel.setControlText(Client.panelRecovery, Client.controlRecovery2,
+					"@yel@your security questions. You will then be able to reset your password");
+
+			for (int i = 0; i < 5; ++i) {
+				Panel.setControlText(Client.panelRecovery, Client.controlRecoveryInput[i], "");
+			}
+
+			Panel.setControlText(Client.panelRecovery, Client.recoverOldPassInput, "");
+			Panel.setControlText(Client.panelRecovery, Client.recoverNewPassInput, "");
+			Panel.setControlText(Client.panelRecovery, Client.recoverConfirmPassInput, "");
+			return;
+		} catch (Exception e) {
+			Client.setLoginMessage("Check your internet settings", "Sorry! Unable to connect to server.");
+			return;
+		}
+  }
+  
+  public static void recover() {
+	  Client.setLoginMessage("Connecting to server", "Please wait...");
+
+		try {
+			int port = Replay.connection_port;

```

Comment:
not necessary to declare this?...

Reply:
probably could pass in directly the argument to initialize stream, but felt it might line up better putting it similar to how is in register